front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -431,33 +434,87 @@ const VaultDataSourceViewItem = ({
   item,
   owner,
   vault,
+  node,
 }: {
   item: DataSourceViewType;
   owner: LightWorkspaceType;
   vault: VaultType;
+  node?: DataSourceViewContentNode;
 }): ReactElement => {
   const router = useRouter();
+  const [isExpanded, setIsExpanded] = useState(false);
 
-  const LogoComponent = getConnectorProviderLogoWithFallback(
-    item.dataSource.connectorProvider,
-    FolderIcon
-  );
-  const dataSourceViewPath = `/w/${owner.sId}/vaults/${vault.sId}/categories/${item.category}/data_source_views/${item.sId}`;
+  const { isNodesLoading, nodes } = useDataSourceViewContentNodes({
```

Comment:
To avoid having all this conditional code (all the `node ? x : y`), I'd suggest doing another dedicated component.

Reply:
Not really sure it's really useful, there are only 3 places where we test "node" - all the rest of logic and rendering is exactly the same

Reply:
Up to you 🙃